### PR TITLE
[test] Add unit test for PublicKeyStore

### PR DESCRIPTION
**Pull request by OlivierChirouze**
And round timestamps to the closest integer

### CODE REVIEW

**Comment by OlivierChirouze** [comment]
Please note that I am aware I did mix a couple of **fixes** with unit tests, which is not ideal.
Sorry about that, it was much more practical for me to push them on the same commit, even if I realize it is not the cleanest option.

File: paf-mvp-operator-client-express/src/operator-client.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export class OperatorClient {
   buildPreferences(
     identifiers: Identifiers,
     data: { use_browsing_for_personalization: boolean },
-    timestamp = new Date().getTime()
+    timestamp = getTimeStampInSec()
```
Comment (OlivierChirouze) [inline paf-mvp-operator-client-express/src/operator-client.ts:51]: bugfix: before timestamp was in milliseconds!

File: paf-mvp-operator-express/src/operator-api.ts [hunk 2]
```diff
@@ -217,7 +217,7 @@ export const addOperatorApi = (
 
       res.send(signedData);
     } catch (e) {
-      res.sendStatus(400);
+      res.status(400);
```
Comment (OlivierChirouze) [inline paf-mvp-operator-express/src/operator-api.ts:220]: bugfix! The exception was not sent before

File: paf-mvp-operator-express/src/operator-api.ts [hunk 3]
```diff
@@ -294,7 +294,7 @@ export class OperatorApi {
     )
   ) {}
 
-  generateNewId(timestamp = new Date().getTime()): Identifier {
+  generateNewId(timestamp = getTimeStampInSec()): Identifier {
```
Comment (OlivierChirouze) [inline paf-mvp-operator-express/src/operator-api.ts:297]: bugfix: before timestamp was in milliseconds!

File: paf-mvp-core-js/src/express/key-store.ts [hunk 4]
```diff
@@ -1,26 +1,31 @@
...
-    const nowTimestampSeconds = new Date().getTime() / 1000;
+    const nowTimestampSeconds = this.timestampProvider();
 
     const existingKey = this.cache[domain];
 
     // Make sure this key is not out dated. If so, then consider no cache value and request it from identity endpoint
-    if (existingKey && nowTimestampSeconds < existingKey.end.getTime()) {
+    if (
+      existingKey &&
+      (existingKey.endTimestampInSec === undefined || nowTimestampSeconds < existingKey.endTimestampInSec)
```
Comment (OlivierChirouze) [inline paf-mvp-core-js/src/express/key-store.ts:27]: Bugfix!
1. was triggering error if the public key doesn't have an end date
2. was comparing milliseconds with seconds! 😱

**Review by kovtun-d — APPROVED** [review]

**Review by kovtun-d — APPROVED** [review]

**Pull request merged**